tests/components/application-connector/test/compass-runtime-agent/synchronisation_test.go

```diff
@@ -44,13 +48,29 @@ func (cs *CompassRuntimeAgentSuite) TestApplication() {
 	cs.Run("Compass Runtime Agent should create Application", func() {
 		err = cs.appComparator.Compare(cs.T(), expectedAppName, synchronizedCompassAppName)
 		cs.NoError(err)
+
+		correctState = err == nil
 	})
 
 	cs.Run("Update app", func() {
-		_ = cs.updateAndWaitForCompare(applicationInterface, synchronizedCompassAppName, applicationID, updatedAppName)
+		if !correctState {
+			cs.T().Skip("App not in correct state")
+		}
+		_ = cs.updateAndWait(applicationInterface, synchronizedCompassAppName, applicationID)
 
 		err = cs.appComparator.Compare(cs.T(), updatedAppName, synchronizedCompassAppName)
 		cs.NoError(err)
+
+		correctState = err == nil
+	})
+
+	cs.Run("With bundles compare", func() {
```

Comment:
Please consider adding a bundle to the app used for `Compass Runtime Agent should create Application` test. This approach reduces number of GraphQL calls.